### PR TITLE
Add installation script for tracing

### PR DESCRIPTION
**Pull request by rfratto**
This PR does a few things:

1. Add basic support for configuring the Tempo subsystem within the v1 Tanka library.
1. Utilize the tanka configs to expose a sane default config `production/kubernetes/agent-tempo.yaml`.
1. Add a bash script to render environment variables in the default config. (This is inline with the other "install" bash scripts).
1. Rename `remote_write` in the Tempo config to `push_config`. `remote_write` should be reserved for fields that act fully like the Prometheus `remote_write`; an array with the standard fields. Once the Tempo subsystem supports an array of targets to write spans to, a `remote_write` field name can return.

### CODE REVIEW

File: production/kubernetes/build/templates/tempo/main.jsonnet [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        username: '${TEMPO_USERNAME}',
+        password: '${TEMPO_PASSWORD}',
+      },
+      batch: {
+        timeout: '5s',
+        send_batch_size: 1000,
+      },
+      queue: {
+        retry_on_failure: false,
+      },
```
Comment (rfratto) [inline production/kubernetes/build/templates/tempo/main.jsonnet:29]: I'm not 100% sure about if these defaults will work for most people.

File: production/kubernetes/build/templates/tempo/main.jsonnet [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }) +
+    agent.withTempoSamplingStrategies({
+      default_strategy: {
+        type: 'probabilistic',
+        param: 0.001,
+      },
+    }) +
```
Comment (rfratto) [inline production/kubernetes/build/templates/tempo/main.jsonnet:36]: Is this too low for most users?

File: production/kubernetes/install-loki.sh [hunk 3]
```diff
@@ -35,7 +35,7 @@ MANIFEST_URL=${MANIFEST_URL:-https://raw.githubusercontent.com/grafana/agent/${M
 LOKI_USERNAME_SET=0
 LOKI_PASSWORD_SET=0
 
-while getopts "l:u:p:" opt; do
+while getopts "h:u:p:" opt; do
```
Comment (rfratto) [inline production/kubernetes/install-loki.sh:38]: This was a bug in the Loki script. I know it should be its own PR, but I'm being lazy by sneaking in a fix.

**Comment by rfratto** [comment]
If you want to sanity check the output, take a look at [this file](https://github.com/rfratto/agent/blob/tempo-tanka-configs/production/kubernetes/agent-tempo.yaml).

File: production/kubernetes/build/templates/tempo/main.jsonnet (hunk 2 above)
Comment (joe-elliott) [reply production/kubernetes/build/templates/tempo/main.jsonnet:36]: This is Jaeger's default.  It's easy to destroy your backend.  Better to start low and ramp up.

File: production/kubernetes/build/templates/tempo/main.jsonnet (hunk 1 above)
Comment (joe-elliott) [reply production/kubernetes/build/templates/tempo/main.jsonnet:29]: yes, agree i don't know either.
batch timeout should be relatively low to make sure that spans make it to the backend in a timely manner when volume is low.  1000 was chosen semi randomly.  It matches our match accepted batch size in tempo ops.  This config will probably go to prod.

`retry_on_failure` is weird.  It really should be false.   The otlp grpc exporter already retries certain GRPC error codes and I don't know the exact mechanics for when and how the queue processor retries.  Also, generally in tracing, the volume is so high that it's preferred to start ditching spans to load shed then queue up a billion spans and continue retrying.

File: production/kubernetes/README.md [hunk 4]
```diff
@@ -7,11 +7,13 @@ Manifests:
 
 - Metric collection: [`agent.yaml`](./agent.yaml)
 - Log collection: [`agent-loki.yaml`](./agent-loki.yaml)
+- Trace collection: [`agent-tempo.yaml`](./agent-tempo.yaml)
 
 Installation script:
 
 - Metric collection: [`install.sh`](./install.sh)
 - Log collection: [`install-loki.sh`](./install-loki.sh)
+- Tempo collection: [`install-tempo.sh`](./install-tempo.sh)
```
Comment (joe-elliott) [inline production/kubernetes/README.md:16]: should this also say "Trace collection"?

File: production/kubernetes/agent-tempo.yaml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                  - __meta_kubernetes_pod_name
+                target_label: pod
+              - action: replace
+                source_labels:
+                  - __meta_kubernetes_pod_container_name
+                target_label: container
+            tls_config:
+                ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+                insecure_skip_verify: false
+  strategies.json: '{"default_strategy": {"param": 0.001, "type": "probabilistic"}}'
```
Comment (joe-elliott) [inline production/kubernetes/agent-tempo.yaml:52]: i guess this is a good idea?  what is the mechanism for overriding this?  

if the user can't customize this it may make more sense to just leave it out.

File: production/kubernetes/agent-tempo.yaml [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+                password: ${TEMPO_PASSWORD}
+                username: ${TEMPO_USERNAME}
+            batch:
+                send_batch_size: 1000
+                timeout: 5s
+            endpoint: ${TEMPO_ENDPOINT}
+            queue:
+                retry_on_failure: false
+        receivers:
+            jaeger:
```
Comment (joe-elliott) [inline production/kubernetes/agent-tempo.yaml:24]: how do we provide a mechanism for configuring receivers?

the jaeger agent normally starts with thrift compact/thrift binary.

File: production/kubernetes/agent-tempo.yaml (hunk 5 above)
Comment (rfratto) [reply production/kubernetes/agent-tempo.yaml:52]: If they're running with the install script the only real option is manually editing the config map, but if they're using the Tanka configs they'd override it [here](https://github.com/rfratto/agent/blob/f484ff5190f8be994c928646447e7c52813dc3ab/production/kubernetes/build/templates/tempo/main.jsonnet#L31-L36).

File: production/kubernetes/agent-tempo.yaml (hunk 6 above)
Comment (rfratto) [reply production/kubernetes/agent-tempo.yaml:24]: IMO the install script should be extremely opinionated, we can change the defaults though. If users want anything else they're free to manually edit the manifest, generate their own (with `make example-kubernetes`), or use the Tanka library.

I can add `thrift_binary` into the mix.

**Comment by rfratto** [comment]
I just noticed I'm not exposing any ports for the Jaeger receiver on the service (or pod). I'll add that into the mix.

**Review by joe-elliott — APPROVED** [review]
